pallets/avn-proxy/src/tests/proxy_tests_with_fees.rs

```diff
@@ -120,16 +119,19 @@ mod charging_fees {
                 let relayer_balance = Balances::free_balance(context.relayer.account_id());
 
                 // Dispatch fails
-                assert_err!(
-                    AvnProxy::proxy(
-                        Origin::signed(context.relayer.account_id()),
-                        inner_call,
-                        payment_authorisation
-                    ),
-                    NftManagerError::<TestRuntime>::ExternalRefIsMandatory
-                );
+                assert_ok!(AvnProxy::proxy(
+                    Origin::signed(context.relayer.account_id()),
+                    inner_call,
+                    payment_authorisation
+                ));
 
-                // No success events emitted
+                assert_eq!(
+                    true,
+                    proxy_failed_inner_dispatch_event_emitted(
+                        context.relayer.account_id(),
+                        call_hash
```

Comment:
In the previous version, we had an error indicating the source of the failure. Do we lose that with the new event? Is there any way we could wrap it in the dispatch failed event, or emit both for example?

Reply:
We can keep it, but I intentionally dropped it.
What is more valuable i think in the context of avn-proxy is matching that the InnerCallFailed event has been emited, that contains the DispatchError information.
Makes the tests more generic and testing the proxy behaviour.
In the staking pallet we do the inverted matching.

Reply:
I think the tests  would be more robust if they tested for the specific error case.

Reply:
Yes @alexpArtos, exactly that.